livekit/src/room/participant/local_participant.rs

```diff
@@ -161,7 +161,7 @@ impl LocalParticipant {
         tokio::spawn({
             let rtc_engine = self.inner.rtc_engine.clone();
             async move {
-                let _ = rtc_engine.negotiate_publisher().await;
+                let _ = rtc_engine.publisher_negotiation_needed().await;
```

Comment:
No this function doesn't wait the negotiation, the negotiation is always async because it is now debounced.
Should all calls to this function wait for the result?

Reply:
`publisher_negotiation_needed` is used to notify the rtc_engine that a negotiation is required to apply new changes, and we can't wait for it, everything happens transparently inside the rtc_session

Reply:
Just pushed a new commit, maybe this is easier to understand now

Reply:
yeah I don't think it should wait for the result. in the publishing track, that's the only case it should wait for the negotiation to be completed.